packages/SwingSet/src/liveslots/collectionManager.js

```diff
@@ -129,6 +129,21 @@ export function makeCollectionManager(
   // TODO Should we be using the new encodeBigInt scheme instead, anyway?
   const BIGINT_TAG_LEN = 10;
 
+  function deleteCollectionEntry(collectionID, vobjID) {
+    const ordinalKey = prefixc(collectionID, `|${vobjID}`);
+    const ordinalString = syscall.vatstoreGet(ordinalKey);
+    syscall.vatstoreDelete(ordinalKey);
+    const ordinalTag = zeroPad(ordinalString, BIGINT_TAG_LEN);
+    const recordKey = prefixc(collectionID, `r${ordinalTag}:${vobjID}`);
+    const rawValue = syscall.vatstoreGet(recordKey);
+    if (rawValue !== undefined) {
+      const value = JSON.parse(rawValue);
+      value.slots.map(vrm.removeReachableVref);
```

Comment:
Or, since it looks like the function that calls this is prepared to accept a `doMore` return value, maybe just improve the `map` to track the `removeReachableVref` return value in the same way that #5055 does it, and return that here. Then we can put off the cleanup for later, knowing that this does the right thing.

Reply:
Also.. this is probably a place where the typechecker would have helped, since the caller is expecting a boolean and this is returning void.

Reply:
Good catch!

I added JSDoc comments to the function, though this is not the complete solution since the type info also needs to be propagated through `setDeleteCollectionEntry` et al, which is far more TS pain than I care to handle right now.